### ISSUE TITLE
1093 You can't specify target table 'cplaix' for update in FROM clause

### ISSUE THREAD

**Issue by svdockum** [dweeves/magmi-git#396]
For some reason, after upgrading to PHP7 on a new server, i get this error on the sku of the product import list.

1093 You can't specify target table 'cplaix' for update in FROM clause

Looking at the error its a sql 'updating the same table error'... but how to fix it?

**Comment by dcarl-dotsource** [comment]
This is related to a change in MySQL 5.7 http://stackoverflow.com/questions/33729296/mysql-5-7-error-1093-you-cant-specify-target-table-for-update-in-from-cla.

I've changed the query in [magmi/plugins/base/itemprocessors/related/related_products.php](https://github.com/dweeves/magmi-git/blob/892b7cebe5c7756ed358a0d28977e0c85780e540/magmi/plugins/base/itemprocessors/related/related_products.php#L248) in `afterImport` to
```
$sql = <<<SQL
DELETE a FROM $cplai AS a
INNER JOIN (
    SELECT value_id
    FROM $cplai
    GROUP BY link_id
    HAVING value_id != MAX(value_id)
) AS b ON a.value_id = b.value_id
SQL;
```
This is valid, but I'm note sure if it really does the same as the current query and I don't have the time to dig deeper into it and test it.

**Comment by RWojenkowski** [comment]
Unfortunately @dcarl-dotsource answer seems to be incorrect, at least for me. Once again it groups by `link_id` but selects `value_id` and MySQL 5.7 gives similar error to the original one.

Has anyone got other idea how to fix this?

**Comment by dcarl-dotsource** [comment]
@RWojenkowski are you shure that the error is caused by this query? There are similar queries still in the code that could cause the error.
Selecting values that are not part of the group by clause is only an issue is you have set [ONLY_FULL_GROUP_BY](http://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_only_full_group_by).

**Comment by RWojenkowski** [comment]
Yes, that's exactly my problem, ONLY_FULL_GROUP_BY has to enabled on our servers. I've got full trace of the error so I know exactly that the problem is with this query when ONLY_FULL_GROUP_BY is enabled.

**Comment by RWojenkowski** [comment]
Our fix is:
```
DELETE FROM $cplai WHERE value_id IN
 		  (SELECT cplai2.value_id FROM
            (SELECT cplai.link_id, MAX(cplai.value_id) latest FROM $cplai cplai GROUP BY cplai.link_id) cplai1
            JOIN (SELECT cplai.link_id, cplai.value_id FROM $cplai cplai) cplai2
            ON cplai1.link_id = cplai2.link_id WHERE cplai1.latest != cplai2.value_id)
```
